integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -169,21 +169,33 @@ def get_root_groups(self) -> List[Group]:
         return typing.cast(
             List[Group], [group for group in groups if group.parent_id is None]
         )
+    
+    def get_specified_groups(self, groups_paths) -> List[Group]:
+        groups = self.gitlab_client.groups.list(iterator=True)
+        return typing.cast(
+            List[Group], [group for group in groups if group.full_path in groups_paths]
+        )
 
-    def create_webhooks(self) -> list[int | str]:
-        root_partial_groups = self.get_root_groups()
-        logger.info("Getting all the root groups to create webhooks for")
-        # Filter out root groups that are not in the group mapping and creating webhooks for the rest
+    # 1. create webhooks by groups 2. create webhooks from root 
+    def create_webhooks(self, groups_paths) -> list[int | str]:
+        if(groups_paths == None):
+            partial_groups = self.get_root_groups()
+            logger.info("Getting all the root groups to create webhooks for")
```

Comment:
can you fix this log pls? i know it was broken before
also move this log before the action itself same goes for the else

Reply:
what do you mean by broken? the grammar?